packages/opal-server/opal_server/git_fetcher.py

```diff
@@ -47,8 +57,24 @@ async def _pull(self):
             repo.references[f"refs/remotes/origin/{self._source.branch}"].resolve().name
         )
 
-    async def make_bundle(self) -> PolicyBundle:
-        pass
+    def make_bundle(self, base_hash: Optional[str] = None) -> PolicyBundle:
+        repo = Repo(str(self._repo_path))
+
+        bundle_maker = BundleMaker(
+            repo,
+            {Path(p) for p in self._source.directories},
+            extensions=self._source.extensions,
+            manifest_filename=self._source.manifest,
+        )
+
+        if not base_hash:
+            return bundle_maker.make_bundle(repo.head.commit)
+        else:
+            try:
+                old_commit = repo.commit(base_hash)
+                return bundle_maker.make_diff_bundle(old_commit, repo.head.commit)
+            except ValueError:
+                raise BadCommitError(base_hash)
```

Comment:
maybe a better fallback would be to just return a full bundle instead of delta bundle?
the client will not verify if it expects a delta bundle.